Mathlib/Combinatorics/SimpleGraph/Connectivity/WalkCounting.lean

```diff
@@ -259,6 +261,39 @@ lemma odd_card_iff_odd_components [Finite V] : Odd (Nat.card V) ↔
   rw [Nat.card_eq_fintype_card, Fintype.card_ofFinset]
   exact (Finset.odd_sum_iff_odd_card_odd (fun x : G.ConnectedComponent ↦ Nat.card x.supp))
 
+lemma odd_components_card_mono [Fintype V] [DecidableEq V] {G' : SimpleGraph V}
+    [DecidableRel G.Adj] (h : G ≤ G') : ({c : ConnectedComponent G' | Odd (Nat.card c.supp)}).ncard
+    ≤ ({c : ConnectedComponent G | Odd (Nat.card c.supp)}).ncard := by
+  have aux (c : G'.ConnectedComponent) (hc : Odd (Nat.card c.supp)) :=
+    Set.nonempty_of_ncard_ne_zero (by
+      intro h'
+      have := (c.odd_card_supp_iff_odd_subcomponents _ h).mp hc
+      rw [h'] at this
+      contradiction
+      : Nat.card {c' : G.ConnectedComponent | c'.supp ⊆ c.supp ∧ Odd (Nat.card c'.supp)} ≠ 0)
```

Comment:
That's a rather odd proof layout. What about
```suggestion
  have aux (c : G'.ConnectedComponent) (hc : Odd (Nat.card c.supp)) :
      {c' : G.ConnectedComponent | c'.supp ⊆ c.supp ∧ Odd (Nat.card c'.supp)}.Nonempty := by
    refine Set.nonempty_of_ncard_ne_zero fun h' ↦ ?_
    simpa [h'] using (c.odd_card_supp_iff_odd_subcomponents _ h).mp hc
```

Reply:
This was kind of what I was looking for, but I didn't arrive at the refine with the inline lambda to make it clean, thanks!

Reply:
Also had some trouble getting it to work with simpa, but by excluding a couple of lemma's it worked out:)